libnetwork/driver_plugin.py

```diff
@@ -348,10 +352,8 @@ def create_endpoint():
 
     app.logger.info("Creating endpoint %s", endpoint_id)
 
-    # Extract relevant data from the Network data.
-    network_data = get_network_data(network_id)
-    gateway_cidr4, _ = get_gateway_pool_from_network_data(network_data, 4)
-    gateway_cidr6, _ = get_gateway_pool_from_network_data(network_data, 6)
+    # Raise error if they don't have network data stored for this network
+    get_network_data(network_id)
```

Comment:
I'll admit I don't fully understand why removing it causes failures. Maybe our tests are just being proactive and checking a situation that we shouldn't care to prevent - if we're creating an endpoint on a network that calico-etcd data doesn't know about